File: OOB class/proj2_java/src/symulator/zwierzeta/czlowiek.java
package symulator.zwierzeta;

import symulator.Swiat;
import symulator.Zwierze;

import java.io.FileWriter;
import java.io.IOException;

public class czlowiek extends Zwierze {
    private int sila_flat, cooldown_ult;

    public czlowiek(Swiat swiat, int ox, int oy, int wiek, int BOARD_WIDTH, int BOARD_HEIGHT) {
        super(swiat, ox, oy,ox, oy, wiek, BOARD_WIDTH, BOARD_HEIGHT);

        this.nazwa = "Czlowiek";
        this.inicjatywa = 4;
        this.sila = 5;
        this.sila_flat = 5;
        this.cooldown_ult = 0;
        this.znak = 'c';
        swiat.setJesliWolnePole(ox, oy, znak);
    }

    @Override
    protected void wykonanie_porodu(int x, int y) {
        swiat.addOrganizm(new czlowiek( swiat,x, y, swiat.get_and_incrementKtoryUrodzony(),this.BOARD_WIDTH,this.BOARD_HEIGHT));
    }



    public czlowiek(Swiat swiat, int x, int y,  int xp, int yp,int wiek, int sila, int sila_flat, int cooldown_ult,  int cp, int BOARD_WIDTH, int BOARD_HEIGHT) {
        super(swiat, x, y,xp,yp, wiek, BOARD_WIDTH, BOARD_HEIGHT);

        this.nazwa = "Czlowiek";
        this.inicjatywa = 4;
        this.sila = sila;
        this.sila_flat = sila_flat;
        this.cooldown_ult = cooldown_ult;
        this.znak = 'c';
        this.cooldown_porodowy = cp;
    }

    @Override
    public void akcja() {
        if (cooldown_porodowy>0)
            cooldown_porodowy--;
        if (sila > sila_flat) {
            sila--;
            if (sila == sila_flat)
                cooldown_ult = 5;
        }
        else if (cooldown_ult > 0)
            cooldown_ult--;

        swiat.rysujSwiat();
        swiat.wypisz_dziennik();
        poprzednie_polozenie[OX] = polozenie[OX];
        poprzednie_polozenie[OY] = polozenie[OY];


        try {
            swiat.getOkno().Czekanie();
        } catch (InterruptedException e) {
            throw new RuntimeException(e);
        }
        swiat.czysc_dziennik();
        swiat.zwolnijPole( poprzednie_polozenie[OX], poprzednie_polozenie[OY]);
    }

    public void ult(){
        if (sila == sila_flat && cooldown_ult == 0) {
            sila += 5;
            swiat.rysujSwiat();
            swiat.wypisz_dziennik();
        }
    }


    public String czlowiek_statusr() {
        if(cooldown_ult==0 && sila>sila_flat)
           return "Umiejetnosc specjalna aktywna\n";
        else if(sila==sila_flat && cooldown_ult!=0)
            return "Umiejetnosc specjalna dostepna za " + cooldown_ult + "\n";
        else
            return "Umiejetnosc specjalna gotowa do uzycia\n";
    }

    @Override
    public void zwiekszSile(int oIle) {
        this.sila += oIle;
        this.sila_flat += oIle;
    }

    @Override
    public void delOrganizm() {
        super.delOrganizm();
        swiat.setCzlowiek(null);
        swiat.rysujSwiat();
        try {
            swiat.getOkno().ekranSmierci();
        } catch (InterruptedException e) {
            throw new RuntimeException(e);
        }
        System.exit(0);
    }

    @Override
    public void zapiszOrganizm(FileWriter zapis) {
        if(sila>sila_flat)
            sila++;
        if(cooldown_ult > 0)
            cooldown_ult++;
        try{
            super.zapiszOrganizm(zapis);
            zapis.write(sila_flat+"");
            zapis.write(" ");
            zapis.write(cooldown_ult+"");
            zapis.write(" ");
            if(sila>sila_flat)
                sila--;
            if(cooldown_ult > 0)
                cooldown_ult--;
        }
        catch(IOException e) {
            System.out.println("An error occurred.");
            e.printStackTrace();
        }

    }

}

File: OOB class/proj1_cpp/czlowiek.cpp
#include "czlowiek.h"
#include "Zwierze.h"
#include "Organizm.h"
#include <conio.h>
#include <fstream>


#define KB_UP 75
#define KB_DOWN 77
#define KB_LEFT 72
#define KB_RIGHT 80
#define KB_ESCAPE 27

enum KIERUNKI {
	POLNOC = 0,
	POLUDNIE = 1,
	WSCHOD = 2,
	ZACHOD = 3,
	SPECIAL = 4
};

czlowiek::czlowiek(short x, short y, Swiat* swiat, short wiek) {
	this->wiek = wiek;
	this->nazwa = "Czlowiek";
	this->inicjatywa = 4;
	this->sila = 5;
	this->sila_flat = 5;
	this->cooldown_ult = 0;
	this->znak = 'c';
	this->polozenie[OX] = x;
	this->polozenie[OY] = y;
	this->swiat = swiat;
	swiat->setJesliWolnePole(x, y, znak);
}

czlowiek::czlowiek(short x, short y, Swiat* swiat, short wiek, short sila, short xp, short yp, short cp, short sila_flat, short cooldown_ult){
	this->wiek = wiek;
	this->nazwa = "Czlowiek";
	this->inicjatywa = 4;
	this->sila = sila;
	this->sila_flat = sila_flat;
	this->cooldown_ult = cooldown_ult;
	this->znak = 'c';
	this->polozenie[OX] = x;
	this->polozenie[OY] = y;
	this->poprzednie_polozenie[OX] = xp;
	this->poprzednie_polozenie[OY] = yp;
	this->cooldown_porodowy = cp;
	this->swiat = swiat;
	swiat->setJesliWolnePole(x, y, znak);
}

void czlowiek::wykonanie_porodu(short x, short y) {
	swiat->addOrganizm(new czlowiek(x, y, swiat, swiat->get_and_incrementKtoryUrodzony()));
}

void czlowiek::akcja() {
	if (cooldown_porodowy)
		cooldown_porodowy--;
	swiat->rysujSwiat();
	swiat->wypisz_dziennik();

	if (sila > sila_flat) {
		sila--;
		if (sila == sila_flat)
			cooldown_ult = 5;
	}
	else if (cooldown_ult > 0)
		cooldown_ult--;
	swiat->zwolnijPole(polozenie[OX], polozenie[OY]);
	int direction=5, tmp;
	while (direction > 3) {
		tmp = _getch();
		switch (tmp)
		{
		case KB_LEFT:
			direction = ZACHOD;
			break;

		case KB_RIGHT:
			direction = WSCHOD;
			break;

		case KB_UP:
			direction = POLNOC;
			break;

		case KB_DOWN:
			direction = POLUDNIE;
			break;
		case 's':
			if (sila == sila_flat && cooldown_ult == 0) {
				sila += 5;
				swiat->rysujSwiat();
				swiat->wypisz_dziennik();
			}
			break;
		case 'z':
			swiat->zapiszGre();
			break;
		case 'w':
			swiat->wczytajGreFun();
			direction = POLNOC;
			break;
		case KB_ESCAPE:
			swiat->zakonczGreFun();
			direction = POLNOC;
			break;
		}
	}
	swiat->czysc_dziennik();
	poprzednie_polozenie[OX] = polozenie[OX];
	poprzednie_polozenie[OY] = polozenie[OY];
	wykonanie_akcji(direction);
}

void czlowiek::czlowiek_status() {
	if(cooldown_ult==0 && sila>sila_flat)
		printf("%c  Sila - %d   Umiejetnosc specjalna aktywna\n", 186, sila);
	else if(sila==sila_flat && cooldown_ult!=0)
		printf("%c  Sila - %d   Umiejetnosc specjalna dostepna za %d\n", 186, sila,cooldown_ult);
	else
		printf("%c  Sila - %d   Umiejetnosc specjalna gotowa do uzycia\n", 186, sila, cooldown_ult);
}

void czlowiek::zwiekszSile(short oIle) {
	this->sila += oIle;
	this->sila_flat += oIle;
}

czlowiek::~czlowiek() {
	swiat->zakonczGreFun();
	swiat->setCzlowiek(nullptr);
}

void czlowiek::zapiszOrganizm(std::fstream& zapis) {
	zapis << znak;
	zapis << " ";
	zapis << polozenie[OX];
	zapis << " ";
	zapis << polozenie[OY];
	zapis << " ";
	zapis << sila;
	zapis << " ";
	zapis << wiek;
	zapis << " ";
	zapis << poprzednie_polozenie[OX];
	zapis << " ";
	zapis << poprzednie_polozenie[OY];
	zapis << " ";
	zapis << cooldown_porodowy;
	zapis << " ";
	zapis << sila_flat;
	zapis << " ";
	zapis << cooldown_ult;
	zapis << " ";
}